include/ringmesh/mesh/cartesian_grid.h

```diff
@@ -281,11 +281,126 @@ namespace RINGMesh
          * @return the attribute manager containing the grid point attributes
          * values.
          */
-        GEO::AttributesManager& attributes_manager() const
+        GEO::AttributesManager& attributes_manager()
         {
             return attributes_manager_;
         }
 
+        /*!
+         * @return the attribute manager containing the grid point attributes
+         * values.
+         */
+        const GEO::AttributesManager& attributes_manager() const
+        {
+            return attributes_manager_;
+        }
+
+        /*!
+         * Allows us to avoid code duplication on the attribute creation
+         * function.
+         */
+        template < typename T >
+        static std::string type_to_string()
+        {
+            if( typeid( T ).name() == typeid( index_t ).name() )
+            {
+                return "index_t";
+            }
+            else if( typeid( T ).name() == typeid( int ).name() )
+            {
+                return "int";
+            }
+            else if( typeid( T ).name() == typeid( float ).name() )
+            {
+                return "float";
+            }
+            else if( typeid( T ).name() == typeid( double ).name() )
+            {
+                return "double";
+            }
+            else
+            {
+                throw RINGMeshException(
+                    "CartesianGrid", "Give a valid attribute type." );
+            }
+        }
+
+        /*!
+         * Creates and adds a new attribute to the cartesian grid from its
+         * name and values on the grid.
+         * \param[in] attribute_name the name of the new attribute to add
+         * \param[in] values a vector of the values of the attribute on all
+         * the cartesian grid cells.
+         */
+        template < typename T >
+        void add_attribute(
+            const std::string& attribute_name, const std::vector< T >& values )
+        {
+            if( values.size() != nb_total_cells_ )
+            {
+                throw RINGMeshException( "CartesianGrid",
+                    "The attribute you're trying to add doesn't have the same"
+                    " number of values than the grid." );
+            }
+            attributes_manager_.bind_attribute_store( attribute_name,
+                GEO::AttributeStore::
+                    create_attribute_store_by_element_type_name(
+                        type_to_string< T >(), 1 ) );
+            GEO::Attribute< T > attribute{ attributes_manager_,
+                attribute_name };
+            for( auto i : range( nb_total_cells_ ) )
+            {
+                attribute[i] = values[i];
+            }
+        }
+
+        /*!
+         * Creates a new attribute with the name \attribute_name and
+         * the same value \value on every grid point.
```

Comment:
Maybe you can change the name to include the fact that it is the same constant value everywhere